client/network/src/service/out_events.rs

```diff
@@ -191,17 +179,16 @@ impl OutChannels {
 	/// Sends an event.
 	pub fn send(&mut self, event: Event) {
 		self.event_streams.retain_mut(|sender| {
-			let queue_size = sender.queue_size.fetch_add(1, Ordering::Relaxed);
-			if queue_size == sender.queue_size_warning && !sender.warning_fired {
+			if sender.inner.len() == sender.queue_size_warning && !sender.warning_fired {
```

Comment:
```suggestion
			if sender.inner.len() >= sender.queue_size_warning && !sender.warning_fired {
```

If this is the only sender for this channel then it shouldn't make a difference, but just in case.

Reply:
Yes, thanks! I forgot that the queue size counting is not atomic now.